### PR TITLE
👔 Update some pipeline plumbing to allow updated rbc-options' derivatives

### PR DESCRIPTION
**Pull request by shnizzedy**
## Fixes

Fixes #1818 & #1821 by @shnizzedy 

### f/alff

Updates [`CPAC/nuisance/nuisance.py`](https://github.com/FCP-INDI/C-PAC/compare/develop...RBC/derivatives#diff-994eca3b014a8a5b0002892e3c488f8a4cad5bc30106be31f786f89d789d541f) to

* update nuisance regression outputs to
  * set `desc-preproc_bold`, `desc-cleaned_bold`, and `desc-denoisedNofilt_bold` all to the same output of `nuisance_regression` if no bandpass (adding the farthest-to-the-right red line in the figure below)
  * keep the same connections as before for `desc-preproc_bold`, `desc-cleaned_bold`, and `desc-denoisedNofilt_bold` if `['bandpass_filtering_order'] == 'Before'`
  * run an extra round of nuisance regression to generate `desc-denoisedNofilt_bold` if `['bandpass_filtering_order'] == 'Before'` (the dashed blue line in the figure)
  <span name="figure"></span>
  `desc-denoisedNofilt_bold`,
* move `warp_denoiseNofilt_to_T1template` in `cpac_pipeline.py` to after `nuisance_regression`, and
* update the condition to run `warp_denoiseNofilt_to_T1template` only if we're not running nuisance regression in template space

### reho

Updates input `space-template_desc-bold_mask` → `space-template_res-derivative_desc-bold_mask`

### network_centrality

Sets https://github.com/FCP-INDI/C-PAC/blob/c55112cab900a46550286fc4e788bb3afc2112ec/CPAC/resources/configs/pipeline_config_rbc-options.yml#L245

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the `develop` branch of the repository. 
- [x] My commit messages follow best practices.
- [x] My code follows the established code style of the repository.
- [ ] I added tests for the changes I made (if applicable).
- [ ] I updated the changelog.
- [ ] I added or updated documentation (if applicable).
- [ ] I tried running the project locally and verified that there are no visible errors (in progress).

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: CPAC/nuisance/nuisance.py [hunk 1]
```diff
@@ -2450,9 +2450,16 @@ def nuisance_regression(wf, cfg, strat_pool, pipe_num, opt, space):
...
+    bandpass = 'Bandpass' in opt
+    bandpass_before = bandpass and cfg['nuisance_corrections',
+                                       '2-nuisance_regression',
+                                       'bandpass_filtering_order'] == 'Before'
+
     nuis = create_nuisance_regression_workflow(opt, name='nuisance_regression'
                                                f'_{space}_{opt["Name"]}'
                                                f'_{pipe_num}')
+    if bandpass_before:
+        nofilter_nuis = nuis.clone(name=f'{nuis.name}-noFilter')
```
Comment (shnizzedy) [inline CPAC/nuisance/nuisance.py:2462]: Is [this](https://github.com/FCP-INDI/C-PAC/pull/1824#figure) a reasonable thing to do? Ie, is it reasonable to run nuisance twice in the bandpass-before case to get a non-filtered bold image?

File: CPAC/pipeline/cpac_pipeline.py [hunk 2]
```diff
@@ -1386,13 +1387,16 @@ def build_workflow(subject_id, sub_dict, cfg, pipeline_name=None,
             pipeline_blocks += [alff_falff]
 
     if 'Template' in target_space_alff:
+        if not nuisance_template and not rpool.check_rpool(
+                'space-template_desc-denoisedNofilt_bold'):
```
Comment (shnizzedy) [inline CPAC/pipeline/cpac_pipeline.py:1391]: Why doesn't `rpool` have `space-template_desc-denoisedNofilt_bold` if we're running https://github.com/FCP-INDI/C-PAC/blob/4cc96f35874cce8d5e6175029be0b1a913b439a2/CPAC/nuisance/nuisance.py#L2592-L2621 ?

File: CPAC/registration/registration.py [hunk 3]
```diff
@@ -4267,10 +4265,8 @@ def warp_deriv_mask_to_T1template(wf, cfg, strat_pool, pipe_num, opt=None):
      "switch": [["registration_workflows", "functional_registration",
                 "func_registration_to_template", "run"],
                 ["registration_workflows", "anatomical_registration", "run"]],
-     "option_key": ["registration_workflows", "functional_registration",
-                    "func_registration_to_template", "apply_transform",
-                    "using"],
-     "option_val": "default",
+     "option_key": "None",
+     "option_val": "None",
```
Comment (shnizzedy) [inline CPAC/registration/registration.py:4269]: Is this okay? Or should these be applied differently if not

```YAML
registration_workflows:
  functional_registration:
    func_registration_to_template:
      apply_transform:
        using: default
```

?

File: CPAC/registration/registration.py [hunk 4]
```diff
@@ -4238,10 +4238,8 @@ def warp_bold_mask_to_T1template(wf, cfg, strat_pool, pipe_num, opt=None):
      "switch": [["registration_workflows", "functional_registration",
                  "func_registration_to_template", "run"],
                 ["registration_workflows", "anatomical_registration", "run"]],
-     "option_key": ["registration_workflows", "functional_registration",
-                    "func_registration_to_template", "apply_transform",
-                    "using"],
-     "option_val": "default",
+     "option_key": "None",
+     "option_val": "None",
```
Comment (shnizzedy) [inline CPAC/registration/registration.py:4242]: [placeholder]

File: CPAC/reho/reho.py [hunk 5]
```diff
@@ -156,12 +154,12 @@ def reho_space_template(wf, cfg, strat_pool, pipe_num, opt=None):
      "switch": ["run"],
      "option_key": "None",
      "option_val": "None",
-     "inputs": [["space-template_desc-cleaned_bold", "space-template_desc-brain_bold",
+     "inputs": [["space-template_desc-cleaned_bold",
+                 "space-template_desc-brain_bold",
                  "space-template_desc-preproc_bold", "space-template_bold"],
```
Comment (shnizzedy) [inline CPAC/reho/reho.py:159]: Should these also be `space-template_res-derivative_*`?

File: CPAC/pipeline/cpac_pipeline.py (hunk 2 above)
Comment (sgiavasis) [reply CPAC/pipeline/cpac_pipeline.py:1391]: I think this just wasn't added when it should have been - this is needed here.

File: CPAC/registration/registration.py (hunk 3 above)
Comment (sgiavasis) [reply CPAC/registration/registration.py:4269]: I would leave these as `default` as single-step has its own bold mask warping. We do need it for the derivative-resolution bold mask warping however, which I can add easily in my upcoming PR.

File: CPAC/reho/reho.py (hunk 5 above)
Comment (shnizzedy) [reply CPAC/reho/reho.py:159]: 90f9ea72e

File: CPAC/registration/registration.py (hunk 3 above)
Comment (shnizzedy) [reply CPAC/registration/registration.py]: 196dea240

File: CPAC/registration/registration.py (hunk 4 above)
Comment (shnizzedy) [reply CPAC/registration/registration.py]: 196dea240

File: CPAC/pipeline/cpac_pipeline.py (hunk 2 above)
Comment (shnizzedy) [reply CPAC/pipeline/cpac_pipeline.py:1391]: What I was confused about here is how I was getting C-PAC complaining about not having `space-template_desc-denoisedNofilt_bold` in the resource pool after

https://github.com/FCP-INDI/C-PAC/blob/d1d06a53c77bdb5848b666b882bff0298394100f/CPAC/pipeline/cpac_pipeline.py#L1347-L1349

before I added `and not rpool.check_rpool('space-template_desc-denoisedNofilt_bold'):` to the condition in https://github.com/FCP-INDI/C-PAC/blob/d1d06a53c77bdb5848b666b882bff0298394100f/CPAC/pipeline/cpac_pipeline.py#L1389-L1392

but maybe I just made a mistake testing it. Either way, it seems to work okay with this `if` block in place.

**Review by shnizzedy — COMMENTED** [review]
Here are the examples of where I got the idea to do like this:

https://github.com/FCP-INDI/C-PAC/blob/9586d7146eff224200e18622523a56da6a9826a6/CPAC/func_preproc/func_preproc.py#L1082
https://github.com/FCP-INDI/C-PAC/blob/9586d7146eff224200e18622523a56da6a9826a6/CPAC/func_preproc/func_preproc.py#L1158
https://github.com/FCP-INDI/C-PAC/blob/9586d7146eff224200e18622523a56da6a9826a6/CPAC/func_preproc/func_preproc.py#L1183
https://github.com/FCP-INDI/C-PAC/blob/9586d7146eff224200e18622523a56da6a9826a6/CPAC/func_preproc/func_preproc.py#L1218
https://github.com/FCP-INDI/C-PAC/blob/9586d7146eff224200e18622523a56da6a9826a6/CPAC/func_preproc/func_preproc.py#L1242
https://github.com/FCP-INDI/C-PAC/blob/9586d7146eff224200e18622523a56da6a9826a6/CPAC/registration/registration.py#L1931
https://github.com/FCP-INDI/C-PAC/blob/9586d7146eff224200e18622523a56da6a9826a6/CPAC/registration/registration.py#L2016
https://github.com/FCP-INDI/C-PAC/blob/9586d7146eff224200e18622523a56da6a9826a6/CPAC/registration/registration.py#L1931
https://github.com/FCP-INDI/C-PAC/blob/9586d7146eff224200e18622523a56da6a9826a6/CPAC/vmhc/vmhc.py#L24

**Review by sgiavasis — APPROVED** [review]

**Pull request merged**